### PR TITLE
Fixed country name string compare to work with unicode

### PR DESCRIPTION
**Pull request by 8ctopus**
Hi there,

I found a significant issue in the library when looking up for a country that contains non ASCII characters such as Côte d'Ivoire.
The current function to case-insensitive string compare does only work in the ASCII range but not with unicode.
More information on the issue can be found here for example: https://stackoverflow.com/questions/5473542/case-insensitive-string-comparison

I've fixed it and added a test for it.

### CODE REVIEW

**Comment by 8ctopus** [comment]
Anyone alive on the project?

File: src/ISO3166.php [hunk 1]
```diff
@@ -134,9 +134,12 @@ public function getIterator(): \Generator
...
     private function lookup(string $key, string $value): array
     {
+        $value = mb_strtolower($value);
+        
         foreach ($this->countries as $country) {
-            if (0 === strcasecmp($value, $country[$key])
-            ||  0 === strncasecmp($value, $country[$key], strlen($value))) {
+            $comparison = mb_strtolower($country[$key]);
+            
+            if ($value === $comparison || $value === mb_substr($comparison, 0, mb_strlen($value)) {
```
Comment (alcohol) [inline src/ISO3166.php]:
```suggestion
            if ($value === $comparison || $value === mb_substr($comparison, 0, mb_strlen($value))) {
```

**Comment by alcohol** [comment]
Sorry, I have not found much time to work on my open source projects for quite some time now. This change is trivial though and definitely welcome. I'll get it sorted today and get a new release out.

**Comment by alcohol** [comment]
Rebased and merged in 40ab84897f7944566d571ed8745fd5629dd2b566

**Pull request closed without merging**